packages/react-component-library/src/components/Dialog/Dialog.tsx

```diff
@@ -53,16 +55,24 @@ export const Dialog = React.forwardRef<ModalImperativeHandle, DialogProps>(
     }: DialogProps,
     ref
   ) => {
+    const [isLoading, setIsLoading] = useState(false)
+
     const confirmButton: ButtonProps = {
-      onClick: onConfirm,
+      onClick: (event) => {
+        setIsLoading(true)
+
+        Promise.resolve(onConfirm?.(event)).finally(() => setIsLoading(false))
```

Comment:
I am curious what made you go for this over `async` `await`?

Reply:
It was a response to Sonarcloud but it's clear now it can be solved in a better way.

![image](https://github.com/user-attachments/assets/6dcca4f7-b306-4c9a-be06-94edb0e7d639)